    retval = 0
  rescue => e
    $logger.error(e.message)
  end
  exit(retval)
end
